### ISSUE TITLE
Javascript error while starting

### ISSUE THREAD

**Issue by RSDuck** [jcward/vscode-haxe#36]
I just installed the extension and I wanted to try it on a project which uses Luxe, where I worked using Atom.

But once I started VSCode(and opened a Haxe file to activate the extension) nothing happend. So I opened the Developer Tools here is the error it's reporting:

```
threadService.ts:217[Extension Host] Activating extension `haxedevs.haxe` failed:  Cannot assign to read only property 'haxePath' of object '#<Object>'(anonymous function) @ threadService.ts:217emitTwo @ events.js:100emit @ events.js:185handleMessage @ internal/child_process.js:718channel.onread @ internal/child_process.js:444
threadService.ts:217[Extension Host] Here is the error stack:  TypeError: Cannot assign to read only property 'haxePath' of object '#<Object>'
    at Function.haxe_HaxeConfiguration.update (C:\Users\Kemal\.vscode\extensions\haxedevs.haxe-0.1.0\haxe.js:2523:16)
    at Object.HaxeContext [as __class__] (C:\Users\Kemal\.vscode\extensions\haxedevs.haxe-0.1.0\haxe.js:74:25)
    at Main.main.$hx_exports.activate (C:\Users\Kemal\.vscode\extensions\haxedevs.haxe-0.1.0\haxe.js:1032:11)
    at Function.t._callActivateOptional (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:12:22251)
    at Function.t._callActivate (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:12:22034)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:12:21906
    at Object.m [as _notify] (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:5:19713)
    at Object.W.enter (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:5:23291)
    at n.Class.define._run (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:5:25119)
    at n.Class.define._completed (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:5:24560)
```

I made a build.hxml file from the flow file as described in the readme file.

**Comment by jcward** [comment]
Interesting. What version of Haxe and VSCode are you using?

Note: The line referenced in haxe.js is sourced from HaxeConfiguration.hx, line 31, where we assign haxePath on HaxeConfigurationObject.

@pleclech - any quick ideas about this? I can only assume VSCode made their configuration object read-only at some point??

**Comment by RSDuck** [comment]
I'm using VSCode 1.5.3 together with Haxe 3.2.1

**Comment by pleclech** [comment]
Hey Jeff,

Will take a look tomorrow ;)

**Comment by jcward** [comment]
@pleclech of course, at your leisure, and don't spend too much time on it. If we have to say "only compatible with VSCode 1.X and prior" that's fine.

**Comment by jcward** [comment]
Hey, @pleclech -- there's a PR (#37) for cloning the configuration object to get around the read-only error. Are your updates in the configuration object only necessary for the plugins operation, or was there some reason you wanted to update the configuration object by reference?

BTW, #35 and #36 could be duplicates.

**Comment by pleclech** [comment]
Hello @jcward,

Oups with too many work to be done, i forgot to came back to you, shame on me... sorry ...

Well as far as i remember there's no reason other than lazyness i have updated directly the object.

**Comment by jcward** [comment]
@pleclech - ha, no problem, I've been very busy lately also. Ok, it soudns like the PR solution seems reasonable then. I'll test it, then push it up (hopefully I'll have time today or tomorrow!)

**Comment by pleclech** [comment]
Ok, good luck for the work :)

**Comment by jcward** [comment]
Should be resolved via  58336ed
